compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/CheckIrElementVisitor.kt

```diff
@@ -323,7 +331,7 @@ class CheckIrElementVisitor(
 
     override fun visitFunctionReference(expression: IrFunctionReference) {
         super.visitFunctionReference(expression)
-
+        expression.checkFunction(expression.symbol.owner)
```

Comment:
Is this needed though? Looks like either a function is from sources, in which case it'll be validated in `visitFunction`, or from binaries, in which case it's not modified and needs no validation.

Reply:
> or from binaries, in which case it's not modified and needs no validation.

Unfortunately there are cases where we have to modify functions coming from binaries, e.g., when changing the names of functions with inline class parameters, or when adding default argument stubs to binary classes. Similarly, psi2ir has a failure case where we end up adding a stub function without any type information (when symbol resolution fails with the default case). This is only caught if we verify every function call.